### PR TITLE
[Isolated Regions] [Test] Making EFS, Monitoring, AD and DFSM integ tests executable in us-iso regions

### PR DESCRIPTION
**Pull request by gmarciani**
### Description of changes
Make the following integration tests executable in us-iso regions:
1. 'test_multi_az_create_and_update'
1. 'test_multiple_efs'
1. 'test_efs_compute_az'
1. 'test_monitoring'
1. 'test_ad_integration' 
1. 'test_dynamic_file_systems_update'

### Tests
1. PASSED: 'test_multi_az_create_and_update'
1. PASSED: 'test_multiple_efs'
1. PASSED: 'test_efs_compute_az'
1. PASSED: 'test_monitoring'
1. PASSED: 'test_ad_integration'
1. PASSED: 'test_dynamic_file_systems_update'

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/storage/test_efs/test_efs_compute_az/pcluster.config.yaml
Comment (hanwen-pcluste) [inline tests/integration-tests/tests/storage/test_efs/test_efs_compute_az/pcluster.config.yaml]: After the multi AZ support, we could merge test_efs_compute_az and test_efs_same_az. 
One of the queue should be in `private_subnet_ids[0]`. The other queue should be in `private_subnet_ids[1]`.
The benefits are:
1. The test will only require 2 AZs in the region
2. We reduce a redundant test
3. We don't have a special condition for the isolated region

That said, if you would merge this PR first, I am fine

File: tests/integration-tests/tests/update/test_update/test_multi_az_create_and_update/pcluster_update_1.config.yaml
Comment (hanwen-pcluste) [inline tests/integration-tests/tests/update/test_update/test_multi_az_create_and_update/pcluster_update_1.config.yaml]: Nice to have: The condition should be if a region only has two azs, rather than a hard-coded us-isob-east-1

**Review by hanwen-pcluste — DISMISSED** [review]

File: tests/integration-tests/tests/update/test_update/test_multi_az_create_and_update/pcluster_update_1.config.yaml
Comment (gmarciani) [reply tests/integration-tests/tests/update/test_update/test_multi_az_create_and_update/pcluster_update_1.config.yaml]: I agree, this is just a quick&dirty fix to unblock the testing, but since it seems that we may have more time we could also improve

File: tests/integration-tests/tests/storage/test_efs/test_multiple_efs/pcluster.config.yaml
Comment (EddyMM) [inline tests/integration-tests/tests/storage/test_efs/test_multiple_efs/pcluster.config.yaml]: NIT: Is it possible to apply this logic based on the length of `private_subnet_ids` instead of a specific region?

Comment (gmarciani) [reply tests/integration-tests/tests/storage/test_efs/test_multiple_efs/pcluster.config.yaml]: I agree: will do in next rev

File: tests/integration-tests/tests/ad_integration/test_ad_integration.py
Comment (gmarciani) [reply tests/integration-tests/tests/ad_integration/test_ad_integration.py:358]: False alarm on test code

File: tests/integration-tests/tests/storage/test_efs/test_efs_compute_az/pcluster.config.yaml
Comment (enrico-usai) [inline tests/integration-tests/tests/storage/test_efs/test_efs_compute_az/pcluster.config.yaml:41]: it would be good to find a more general approach to solve this kind of issues (e.g. modify the fixture to extend the private_subnet_ids list by duplicating some subnets in iso regions).

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**
